### ISSUE TITLE
Repl

### ISSUE THREAD

**Issue by spoerri** [Southporter/zlox#1]
The repl is flaky for me. Made a small change (see below) to run under ms windows:

```
$ zig version; git diff HEAD~1; zig build -Doptimize=ReleaseSafe; ./zig-out/bin/zlox.exe
0.13.0
diff --git a/src/main.zig b/src/main.zig
index e24d051..c11547b 100644
--- a/src/main.zig
+++ b/src/main.zig
@@ -47,7 +47,7 @@ pub fn main() !void {
     var gpa = std.heap.GeneralPurposeAllocator(.{}){};
     const allocator = gpa.allocator();

-    var args = std.process.args();
+    var args = try std.process.argsWithAllocator(allocator);
     _ = args.skip();
     const file = args.next();
     const extra = args.next();
> print "a";
a
nil> print "b";
a
nil>
```

**Comment by Southporter** [comment]
Hi @spoerri, thanks for taking a look at the repo. I forgot that std.process.args() only works on unix. I'll update to use the allocator.

As far as the repl goes, that still needs to be fixed. Right now I have it rerun the entire session from top to bottom, which is why it will be flaky. I deviated from the Crafting Interpereters book by not using globals, so I still need to figure out a good way to have the repl work as expected.

I've updated the repo to work on Windows and hopefully got the repl to work a bit better. Let me know if you are running into any more weird behavior.
